controllers/topology/internal/mergepatch/managed_paths_test.go

```diff
@@ -0,0 +1,173 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package mergepatch
+
+import (
+	"testing"
+
+	. "github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	"sigs.k8s.io/cluster-api/controllers/topology/internal/contract"
+)
+
+func Test_getManagedPaths(t *testing.T) {
+	tests := []struct {
+		name string
+		obj  client.Object
+		want []contract.Path
+	}{
+		{
+			name: "Return no paths if the annotation does not exists",
+			obj: &unstructured.Unstructured{
+				Object: map[string]interface{}{},
+			},
+			want: []contract.Path{},
+		},
```

Comment:
Maybe add a test case with an empty annotation and maybe just a bigger one something like
```bash
 kubeadmConfigSpec.clusterConfiguration.apiServer.certSANs,
        kubeadmConfigSpec.clusterConfiguration.imageRepository, kubeadmConfigSpec.initConfiguration.nodeRegistration.criSocket,
        kubeadmConfigSpec.initConfiguration.nodeRegistration.kubeletExtraArgs.eviction-hard,
        kubeadmConfigSpec.joinConfiguration.nodeRegistration.kubeletExtraArgs.cgroup-driver,
        kubeadmConfigSpec.joinConfiguration.nodeRegistration.kubeletExtraArgs.eviction-hard,
        machineTemplate.infrastructureRef.apiVersion, machineTemplate.infrastructureRef.kind,
        machineTemplate.infrastructureRef.name, machineTemplate.infrastructureRef.namespace,
        machineTemplate.metadata.labels.cluster.x-k8s.io/cluster-name, machineTemplate.metadata.labels.topology.cluster.x-k8s.io/owned,
        replicas, rolloutStrategy.rollingUpdate.maxSurge, rolloutStrategy.type, version
```

Reply:
Added, it is kind of overkill for the first layer of unit test - considering we have 4 o 5 levels above - but I don't mind expanding a the existing deep nesting test

Reply:
Ah sorry. Don't ask me how but I somehow missed your pre-existing KCP test.